pkg/kubelet/probe.go

```diff
@@ -35,50 +37,83 @@ import (
 
 const maxProbeRetries = 3
 
-// probeContainer probes the liveness/readiness of the given container.
+// Prober helps to check the liveness/readiness of a container.
+// TODO(yifan): Replace the concrete type with interface later.
+type Prober struct {
+	exec   execprobe.ExecProber
+	http   httprobe.HTTPProber
+	tcp    tcprobe.TCPProber
+	runner dockertools.ContainerCommandRunner
+
+	readinessManager *kubecontainer.ReadinessManager
+	refManager       *kubecontainer.RefManager
+	recorder         record.EventRecorder
+}
+
+// NewProber creates a Prober, it takes a command runner and
+// several container info managers.
+func NewProber(
```

Comment:
Or move it to kubelet/container...

Actually this function does belong to kubelet in some sense... Hopefully if kubelet someday does not import dockertools or rkt directly, we can move them back...